test/powershell/Language/Parser/Conversions.Tests.ps1

```diff
@@ -78,6 +78,10 @@ Describe 'conversion syntax' -Tags "CI" {
             $result -join ";" | Should -Be ($Elements -join ";")
         }
     }
+
+    It 'Should not convert invalid strings to type name' {
+        'int]whatever' -as [type] | Should -Be $null
```

Comment:
Suggesting to add the other report from the issue.
```suggestion
        'int]whatever' -as [type] | Should -Be $null
        [Type] 'int]whatever' | Should -Be $null
```

Reply:
Err, so checked with @SeeminglyScience and while the former should be null, the latter cast should throw an invalid argument error.